mantidproject/mantid
# Add algorithm to aggregate stacks images from wavelength dependend data

Add the algorithm as described in #13839. Earlier, python scripts had been added in #14095.
Once the algorithm is available, it will be used from the tomo GUI (#13467).

**To test:**
- Ensure that the new unit and system tests pass. As this is a purely  I/O algorithm the unit tests don't do any real functional test (would be slow). The system test does.
- Check that the documentation reads well.

Open up the algorithm dialog and set:
- InputPath to the directory where you put the files that (at ISIS)  you can grab a small one from \\olympic\Babylon5\Public\fedemp\IMAT\test_stack_test_larmor_201510_one_angle_smallish
- OutputPath to your favorite new/empty directory.
- Run the algorithm and see that what it generates makes sense.
- It has to add quite a few files so it will take some seconds. The progress bar should indicate progress
- There are some bogus directories and files included in the test data, these should be ignored and you might see warnings about it

More points to bear in mind:
- This produces outputs as fits images. You can open them with imageJ, fv, or with the Mantid Tomo GUI.
- If you set the "Log Level" to information or more, you should see messages like "Saved output aggregated image into: /home/fedemp/test/test_imgg_agg/sum_projection_00001_bands_indices_0_1000"
  by the end of the algorithm run.
- For a more intensive (and slow) test, you can grab these other stacks of images as input: 
  - \\olympic\Babylon5\Public\fedemp\IMAT\test_stack_test_larmor_201510_one_angle
  - \\olympic\Babylon5\Public\fedemp\IMAT\test_stack_test_phantom1_3angles

Fixes #13839.

Release notes: https://github.com/mantidproject/mantid/commit/3d2b31144a1434ea1a842e0300f64b1ab33397ae

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
